### PR TITLE
fuzz: initial integration

### PR DESCRIPTION
**Pull request by manunio**
This PR:
 - Adds initial fuzz_target for image decoders.
  - Adds jazzer.js as dev-dependency.

### CODE REVIEW

File: test/fuzz/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Fuzz Testing
+
+Fuzz testing is:
+
+> An automated software testing technique that involves providing invalid, unexpected, or random data as inputs to a program.
+
+We use coverage guided fuzz testing to automatically discover bugs in `pdf.js`
```
Comment (Snuffleupagus) [inline test/fuzz/README.md:7]: Please note that the "correct" casing of the project name is PDF.js, hence:
```suggestion
We use coverage guided fuzz testing to automatically discover bugs in PDF.js.
```

File: test/fuzz/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+This `fuzz/` directory contains the configuration and the fuzz tests for `pdf.js`.
```
Comment (Snuffleupagus) [inline test/fuzz/README.md:9]:
```suggestion
This `fuzz/` directory contains the configuration and the fuzz tests for PDF.js.
```

File: test/fuzz/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+To generate and run fuzz tests, we use [jazzer.js](https://github.com/CodeIntelligenceTesting/jazzer.js/) library.
+
+## Running a fuzzer
+
+This directory contains fuzzers like for example `jpg.fuzz`. You can run it with:
+
+// generate image decoders
+```sh
+$ gulp imag_decoder
```
Comment (Snuffleupagus) [inline test/fuzz/README.md:18]: That command doesn't exist, hence trying to follow these instructions as-is won't work.
```suggestion
$ gulp image_decoders
```

File: test/fuzz/jbig2.fuzz.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const ignored = ["Cannot read properties", "JBIG2 error"];
+
+function ignoredError(error) {
+  return ignored.some(message => error.message.includes(message));
+}
+
+/**
+ * @param {Buffer} data
+ */
```
Comment (Snuffleupagus) [inline test/fuzz/jbig2.fuzz.js:11]: Sorry, but this isn't correct; this comment applies in the files below as well.

Please note that PDF.js library does *not* support Node.js `Buffer`s, note [this code](https://github.com/mozilla/pdf.js/blob/f4b396f6c88e951174879a52390fa89223326b36/src/display/api.js#L534-L544), which thus (implicitly) applies to the image decoders as well.
```suggestion
/**
 * @param {Uint8Array} data
 */
```

File: test/fuzz/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Fuzz Testing
+
+Fuzz testing is:
+
+> An automated software testing technique that involves providing invalid, unexpected, or random data as inputs to a program.
+
+We use coverage guided fuzz testing to automatically discover bugs in `pdf.js`
+
+This `fuzz/` directory contains the configuration and the fuzz tests for `pdf.js`.
+To generate and run fuzz tests, we use [jazzer.js](https://github.com/CodeIntelligenceTesting/jazzer.js/) library.
```
Comment (Snuffleupagus) [inline test/fuzz/README.md:10]:
```suggestion
To generate and run fuzz tests, we use the [Jazzer.js](https://github.com/CodeIntelligenceTesting/jazzer.js/) library.
```

File: test/fuzz/.eslintrc [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": [
+    "../.eslintrc"
+  ],
+
+  "rules": {
+    "import/no-unresolved": ["error", { "ignore": [".*/build/image_decoders/.*"] }],
```
[the remark this answers is not in the document]
Comment (timvandermeij) [reply test/fuzz/.eslintrc:7]: The LTS unit tests failing was probably due to https://github.com/mozilla/pdf.js/compare/9eeb0b2ee956f2ed9a53ff1501f6cbe0d45010dc..e1bf6e48fd552e9b6969ccc181974bce579990bb#diff-63129533f0fcab1395066c5eecb394e84cc3538b5e016a2f0bb49942825610e6L2212-L2215 which is fixed by the rebase.

**Review by timvandermeij — APPROVED** [review]
Looks good to me with these comments addressed. I have been playing with this locally and it runs just fine, and fortunately it hasn't found any crashes or timeouts so far in my (admittedly relatively short) runs.

**Pull request merged**